server/interpreter.js

```diff
@@ -2085,7 +2097,7 @@ Interpreter.prototype['stepAssignmentExpression'] = function() {
   var node = state.node;
   if (!state.doneLeft_) {
     state.doneLeft_ = true;
-    stack.push({node: node['left'], components: true});
+    this.pushNode_(node['left']).components = true;
```

Comment:
This is not very readable.  Perhaps `compenents` *and* `scope` should be optional parameters to `pushNode_`?